### DISCUSSION TITLE
keyboard controls for DevilutionX

### DISCUSSION THREAD

**Discussion by ikonomov** [diasurgical/DevilutionX#4315] [Ideas]
I just made this [keyboard layout](http://www.keyboard-layout-editor.com/##@_backcolor=%23ffffff&name=keyboard%20controls%20for%20DevilutionX%3B&@_c=%23e4a0db&p=FLAT&f:2%3B&=Esc%0A%0A%0Amain%20menu&_x:1%3B&=F1%0A%0A%0Ahelp&_c=%237cc690%3B&=F2%0A%0A%0Aquick%20save&=F3%0A%0A%0Aquick%20load&_c=%23cccccc%3B&=F4&_x:0.5&c=%2344ccc5%3B&=F5%0A%0A%0Aquick%20spell&=F6%0A%0A%0Aquick%20spell&=F7%0A%0A%0Aquick%20spell&=F8%0A%0A%0Aquick%20spell&_x:0.5&c=%23e2e28d%3B&=F9%0A%0A%0Aquick%20message&=F10%0A%0A%0Aquick%20message&=F11%0A%0A%0Aquick%20message&=F12%0A%0A%0Aquick%20message&_x:0.25&c=%237cc690%3B&=PrtSc%0A%0A%0Ascreen%20capture&_c=%23cccccc%3B&=Scroll%20Lock&_c=%237cc690%3B&=Pause%0A%0A%0Apause%3B&@_y:0.5&c=%23cccccc%3B&=~%0A%60&_c=%23dba767%3B&=!%0A1%0A%0A%0A%0A%0A%0Abelt&=%2F@%0A2%0A%0A%0A%0A%0A%0Abelt&=%23%0A3%0A%0A%0A%0A%0A%0Abelt&=$%0A4%0A%0A%0A%0A%0A%0Abelt&=%25%0A5%0A%0A%0A%0A%0A%0Abelt&=%5E%0A6%0A%0A%0A%0A%0A%0Abelt&=%2F&%0A7%0A%0A%0A%0A%0A%0Abelt&=*%0A8%0A%0A%0A%0A%0A%0Abelt&_c=%23cccccc%3B&=(%0A9&=)%0A0&_c=%2380b9ff%3B&=%2F_%0A-%0A%0A%0A%0A%0A%0Aautomap%20zoom&=+%0A%2F=%0A%0A%0A%0A%0A%0Aautomap%20zoom&_c=%23cccccc&w:2%3B&=Backspace&_x:0.25%3B&=Insert&=Home&=PgUp%3B&@_c=%2380b9ff&w:1.5%3B&=Tab%0A%0A%0Aautomap&_c=%23e4a0db%3B&=Q%0A%0A%0Aquest%20log&_c=%23cccccc%3B&=W&=E&=R&=T&=Y&=U&_c=%23e4a0db%3B&=I%0A%0A%0Ainventory&_c=%23cccccc%3B&=O&_c=%237cc690%3B&=P%0A%0A%0Apause&_c=%23cccccc%3B&=%7B%0A%5B&=%7D%0A%5D&_w:1.5%3B&=%7C%0A%5C&_x:0.25%3B&=Delete&=End&=PgDn%3B&@_w:1.75%3B&=Caps%20Lock&=A&_c=%2344ccc5%3B&=S%0A%0A%0Aquick%20spells&_c=%23cccccc%3B&=D&_c=%237cc690%3B&=F%0A%0A%0Aincrease%20gamma&=G%0A%0A%0Adecrease%20gamma&_c=%23cccccc%3B&=H&=J&=K&_c=%23e4a0db%3B&=L%0A%0A%0Achat%20log&_c=%23cccccc%3B&=%2F:%0A%2F%3B&=%22%0A'&_c=%23e2e28d&w:2.25%3B&=Enter%0A%0A%0Asend%20message%3B&@_c=%237cc690&w:2.25%3B&=Shift%0A%0A%0Aprevent%20movement&=Z%0A%0A%0Azoom&_c=%23cccccc%3B&=X&_c=%23e4a0db%3B&=C%0A%0A%0Acharacter%20info&_c=%23e2e28d%3B&=V%0A%0A%0Agame%20info&_c=%23e4a0db%3B&=B%0A%0A%0Aspellbook&_c=%23cccccc%3B&=N&=M&=%3C%0A,&=%3E%0A.&=%3F%0A%2F%2F&_w:2.75%3B&=Shift&_x:1.25&c=%2380b9ff%3B&=%E2%86%91%0A%0A%0AAutomap%20scroll%3B&@_c=%23cccccc&w:1.25%3B&=Ctrl&_w:1.25%3B&=Win&_c=%23fc9c93&w:1.25%3B&=Alt%0A%0A%0Ahighlight%20items&_c=%23e4a0db&w:6.25%3B&=%0A%0A%0Ahide%20open%20screens&_c=%23cccccc&w:1.25%3B&=Alt&_w:1.25%3B&=Win&_w:1.25%3B&=Menu&_c=%23fc9c93&w:1.25%3B&=Ctrl%0A%0A%0Ahighlight%20items%20toggle&_x:0.25&c=%2380b9ff%3B&=%E2%86%90%0A%0A%0AAutomap%20scroll&=%E2%86%93%0A%0A%0AAutomap%20scroll&=%E2%86%92%0A%0A%0AAutomap%20scroll) for DevilutionX, I thought it could be useful to have it included in the Wiki.  Using this editor it is very easy to change and export the layout to a PNG file.

**Comment by ikonomov** [comment]
If this gets added to a Wiki page it should be the place where we can list all additional shortcuts or key combinations:

- toggle fullscreen with alt+enter
- Shift-click consumables to move them between the belt and the inventory
- Shift-click equipment to equip from or unequip to the inventory or stash
- Shift-click to spend all stat points
- Shift-click to clear readied spell
- Ctrl-click to drop items or move them between the stash and the inventory

**Reply by AJenbo** [reply]
FYI you can edit the wiki, it's community maintained.

**Reply by ikonomov** [reply]
I didn't know!  Thank you

**Reply by ikonomov** [reply]
https://github.com/diasurgical/devilutionX/wiki/Keyboard-Controls

**Reply by AJenbo** [reply]
Probably also mention that most keys kan be rebound from the settings menu.
